### PR TITLE
Use mpsc implementation from embassy for Actor queue

### LINKED ISSUE #31: Configure max message queue per ActorContext
At present, the message queue size of an actor is bound to its type using the MessageQueueSize associated type. Instead, having the size configured per ActorContext would allow tuning memory usage to an applications use case.

### PR DESCRIPTION
**Pull request by lulf**
* Remove the channel implementation no longer in use and replace it with
  the implementation from embassy.
* Remove associated type of Actor for configuring message queue size.
  Instead, use a const generic parameter on the ActorContext, allowing
  the application to choose the message queue size independent of the
  Actor type.

Introduce an ActorHandle trait to avoid parameterizing Address by the
message queue size of the destination Actor.

Fixes #31, #56

### CODE REVIEW

**Comment by lulf** [comment]
@huntc ^^ it's working awesomely!

**Comment by huntc** [comment]
> @huntc ^^ it's working awesomely!

That’s great to see! Well done!

File: device/src/kernel/actor.rs [hunk 1]
```diff
@@ -1,31 +1,28 @@
...
+
+use embassy::{
+    executor::{raw::Task, SpawnError, SpawnToken, Spawner},
+    util::{
+        mpsc::{self, Channel, Receiver, RecvFuture, Sender, WithCriticalSections},
+        DropBomb,
+    },
+};
+
+type ActorMutex = WithCriticalSections;
```
Comment (huntc) [inline device/src/kernel/actor.rs:18]: I’m sure you’ve thought of this, but there’s a NoopMutex available also, which is probably more typical to use with Embassy.

Comment (lulf) [reply device/src/kernel/actor.rs:18]: Ah, indeed. Thanks for the suggestion, that fits better.

**Pull request merged**